pyproject.toml

```diff
@@ -1,3 +1,65 @@
 [build-system]
 requires = ['setuptools>=40.8.0', 'wheel', 'reentry~=1.3', 'fastentrypoints~=0.12']
 build-backend = 'setuptools.build_meta:__legacy__'
+
+[tool.pylint.format]
+max-line-length = 120
+
+[tool.pylint.tool-check]
+generated-members = 'self.exit_codes.*'
+
+[tool.pylint.messages_control]
+disable = [
+    'bad-continuation',
+    'duplicate-code',
+    'locally-disabled',
+    'logging-format-interpolation',
+    'inconsistent-return-statements',
+    'import-outside-toplevel',
+    'no-else-raise',
+    'too-many-arguments',
+    'too-many-ancestors',
+    'too-many-branches',
+    'too-many-locals',
+]
+
+[tool.pylint.basic]
+good-names = [
+    'i',
+    'j',
+    'k',
+    'Run',
+    'UpfData',
```

Comment:
Maybe we should just add `'*Data', '*Calculation', '*WorkChain'` here? It's a bit less strict, but we won't have to tediously maintain this list.

Reply:
I did this, but there might be a bug here. See this build failed: https://github.com/aiidateam/aiida-quantumespresso/pull/562/checks?check_run_id=1192909400 among other things because it contains
```
************* Module aiida_quantumespresso.calculations.pw
aiida_quantumespresso/calculations/pw.py:159:12: C0103: Variable name "HpCalculation" doesn't conform to snake_case naming style (invalid-name)
```
Why would `HpCalculation` here not match the `*Calculation` regex? Many other cases are accepted just fine

Reply:
It seems my testing for whether this works was wrong.. but the only ones that are actually needed are ``'UpfData', 'PwCalculation', 'HpCalculation'``.

Note that this is only for the case where a _variable_ is called like this, i.e. when the factory is used to get the data / calculation / workchain type.

We could override the `variable-rgx` setting to make the regex work, but that's a bit ugly.